docs/src/manuals/sharpbits/stack-overflow-inference.md

```diff
@@ -0,0 +1,95 @@
+# [Stack Overflow during inference](@id stack-overflow-inference)
+
+When working with large probabilistic models in RxInfer, you might encounter a `StackOverflowError`. This section explains why this happens and how to prevent it.
+
+## The Problem
+
+RxInfer uses reactive streams to compute messages between nodes in the factor graph. The subscription to these streams happens recursively, which means:
+
+1. Each node subscribes to its input messages or posteriors
+2. Those input messages may need to subscribe to their own inputs
+3. This continues until all dependencies are resolved
+
+For large models, this recursive subscription process can consume the entire stack space, resulting in a `StackOverflowError`.
+
+## Example Error
+
+When this occurs, you'll see an error message that looks something like this:
+
+```julia
+ERROR: Stack overflow error occurred during the inference procedure. 
+```
+
+## Solution: Limiting Stack Depth
+
+RxInfer provides a solution through the `limit_stack_depth` option in the inference options. This option limits the recursion depth at the cost of some performance overhead.
+
+### How to Use
+
+You can enable stack depth limiting by passing it through the `options` parameter to the `infer` function:
+
+```@example stack-overflow-inference
```

Comment:
Is it possible to use julia code blocks?

I think if I just add "julia" right after the three dashes "`" but before the (at)example then it should use color highlighting?

Test:
```julia @example stack-overflow-inference
using RxInfer
@model function long_state_space_model(y)
    x[1] ~ Normal(mean = 0.0, var = 1.0)
    y[1] ~ Normal(mean = x[1], var = 1.0)
    for i in 2:length(y)
        x[i] ~ Normal(mean = x[i - 1], var = 1.0)
        y[i] ~ Normal(mean = x[i], var = 1.0)
    end
end
data = (y = rand(10000), )
using Test #hide
@test_throws StackOverflowError infer(model = long_state_space_model(), data = data) #hide
results = infer(
    model = long_state_space_model(),
    data = data,
    options = (
        limit_stack_depth = 100, # note the comma
    )
)
```

Reply:
it uses highlighted julia code blocks with `@example stack-overflow-inference`, see [here](https://documenter.juliadocs.org/stable/man/syntax/#reference-at-example), these are julia code blocks that are also being executed during the creation of the documentation page, just `julia` will only highlight but not execute